xwiki-platform-core/xwiki-platform-bridge/src/main/java/org/xwiki/bridge/event/DocumentDeletedEvent.java

```diff
@@ -42,28 +42,34 @@ public class DocumentDeletedEvent extends AbstractDocumentEvent
      * changes.
      */
     private static final long serialVersionUID = 1L;
-    
+
+    /**
+     * The reference of the deleted document.
+     */
+    private final DocumentReference sourceReference;
```

Comment:
Would be cleaner to introduce that (but under the more generic "documentReference" naming) at AbstractDocumentEvent level IMO, it does not make much sense to have only DocumentDeletedEvent expose this.